scripts/release.sh

```diff
@@ -61,7 +61,7 @@ main() {
       REPOSITORY=$(pwd)
       BRANCH=$(git rev-parse --abbrev-ref HEAD)
   else
-      REPOSITORY=${REPOSITORY:-"https://github.com/etcd-io/etcd.git"}
+      REPOSITORY=${REPOSITORY:-"git@github.com:etcd-io/etcd.git"}
```

Comment:
If this change is accepted I will backport it to `release-3.4` and `release-3.5`.

Reply:
> If this change is accepted I will backport it to `release-3.4` and `release-3.5`.

It's OK to backport the change to release-3.5. But let's keep 3.4 as it's. I haven't seen any issues so far when releasing 3.4, we don't really want to any surprise.

Reply:
Fair enough. We haven't seen the issue because we have been manually creating and pushing tag.

Reply:
The scheme/protocol for cloning a repository can be a personal preference. If you have properly configured Git for HTTP authentication, it should work fine. You could potentially use GitHub without an SSH key.

That's why I think the scheme could be opinionated. Maybe a good middle ground would be to receive an option `--use-https-repository-url` (or `--use-ssh-repository-url`), to switch from https and ssh, and keep the release scripts more in sync... Or we could also leave it as-is, according to each release manager preference :)